### ISSUE TITLE
c.lang.security.use-after-free doesn't allow structure member free calls

### ISSUE THREAD

**Issue by paisleyrob** [returntocorp/sgrep-rules#2631]
**Describe the bug**
A structure's member cannot be freed without triggering this rule.

**To Reproduce**
```diff
--- a/c/lang/security/use-after-free.c
+++ b/c/lang/security/use-after-free.c
@@ -80,3 +80,18 @@ int ok_code3(){
     return 0;

 }
+
+struct lv {
+    int length;
+    char *value;
+} lv;
+
+int ok_code4(){
+    int initial = 1000;
+    struct lv lv = {
+        .length = initial,
+        .value = malloc(initial),
+    };
+    free(lv.value);
+    return 0;
+}
```

Causes the following additional output from the following call `semgrep --disable-version-check --config ./c/lang/security/use-after-free.yaml --dataflow-traces c/lang/security/use-after-free.c`

```
     c.lang.security.use-after-free
        Variable 'lv.value' was used after being freed. This can lead to undefined behavior.

         95┆ free(lv.value);

          Taint comes from:
           c/lang/security/use-after-free.c
           95┆ free(lv.value);

          This is how taint reaches the sink:
           c/lang/security/use-after-free.c
           95┆ free(lv.value);
```

**Expected behavior**
Freeing a structure member variable should not trigger a use-after-free warning.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [x] P1: this will block me in the near future
- [ ] P2: annoying but not blocking me

Choice is to either remove the use-after-free call check, or add `// nosem` to the lines that review this.

**Additional Context**
This rule is triggered from the `security-audit.yaml` ruleset: https://semgrep.dev/p/security-audit

**Comment by 0xDC0DE** [comment]
@paisleyrob Thanks for the report! I'll take a look.

**Comment by 0xDC0DE** [comment]
This should be fixed with the next release!
